pkg/image-manager/controller.go

```diff
@@ -114,7 +116,13 @@ func (h *imageEntityHandler) Add(ctx context.Context, obj entitystore.Entity) (e
 
 	defer func() { h.Store.UpdateWithError(ctx, i, err) }()
 
-	if err = h.Builder.imageCreate(ctx, i, &bi); err != nil {
+	if i.Status == entitystore.StatusMISSING {
```

Comment:
`dispatch-server` does not use external registry by default, seems like this will break it. Can we make this behavior configurable through an option?

Reply:
Actually, we should try to pull in every case, and if the error is that image does not exist in the registry, we should re-build it. This will also cover the case where image was deleted both from docker & registry.

Reply:
I think this is what happens i.e if the image is `MISSING` from docker host, we try to pull it and if it does not exists in the Registry as well (or some other error occurred), the image is then moved to the `ERROR` state. This would result in the image getting rebuilt.

The difference is that we do this check if the image exists at a longer interval than before. So if an image is removed from the registry, it will eventually be rebuilt.

Reply:
LGTM